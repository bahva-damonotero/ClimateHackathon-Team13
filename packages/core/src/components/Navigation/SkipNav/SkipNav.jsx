import React, { useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import skipNav from '@uswds/uswds/js/usa-skipnav';

export default function SkipNav({ href }) {
  const linkRef = useRef();

  useEffect(() => {
    const linkEl = linkRef.current;
    if (!linkEl) return;

    // call USWDS's `on` function to initialize the component
    skipNav.on(linkEl);

    // on dismount, call USWDS's `off` function to cleanup the component
    return () => {
      skipNav.off(linkEl);
    };
  }, []);

  function handleClick(e) {
    // prevent router from redirecting - allow uswds js to execute
    e.preventDefault();
  }

  return (
    <a className="usa-skipnav" href={href} ref={linkRef} onClick={handleClick}>
      Skip to main content
    </a>
  );
}

SkipNav.propTypes = {
  /** Anchor link to go to when the Skip Nav link is triggered */
  href: PropTypes.string.isRequired
};
