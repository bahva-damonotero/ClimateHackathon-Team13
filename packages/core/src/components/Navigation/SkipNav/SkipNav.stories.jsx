/* eslint-disable react/prop-types */
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import SkipNav from '.';

export default {
  title: 'Navigation/SkipNav',
  component: SkipNav,
  argTypes: {
    href: '#main-content'
  }
};

const Template = ({ href }) => (
  <MemoryRouter>
    <SkipNav href={href} />
  </MemoryRouter>
);

const defaultArgs = {
  href: '#main-content'
};

export const def = Template.bind({});
def.storyName = 'Default';
def.args = {
  ...defaultArgs
};
