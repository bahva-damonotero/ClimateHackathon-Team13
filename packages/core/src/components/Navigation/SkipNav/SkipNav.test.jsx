import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import skipNavigation from '@uswds/uswds/js/usa-skipnav';

import SkipNav from '.';

describe('SkipNav', () => {
  const href = '#main-content';
  const SKIPNAV_SELECTOR = '.usa-skipnav';

  const jsx = (
    <Router>
      <SkipNav href={href} />
    </Router>
  );

  function getSkipNav(container) {
    return container.querySelector(SKIPNAV_SELECTOR);
  }

  it('should render properly', () => {
    const { container } = render(jsx);

    expect(getSkipNav(container).getAttribute('href')).toBe(href);
    expect(getSkipNav(container)).toBeTruthy();
  });

  it('should call `skipNavigation.on()` to initialize the component', () => {
    const onSpy = jest.spyOn(skipNavigation, 'on');

    render(jsx);

    expect(onSpy).toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    const tree = renderer.create(jsx).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
