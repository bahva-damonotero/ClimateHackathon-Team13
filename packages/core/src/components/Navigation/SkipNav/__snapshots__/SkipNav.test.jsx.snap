// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkipNav should match snapshot 1`] = `
<a
  className="usa-skipnav"
  href="#main-content"
  onClick={[Function]}
>
  Skip to main content
</a>
`;
